modules/terragrunt/options.go

```diff
@@ -0,0 +1,133 @@
+package terragrunt
+
+import (
+	"os"
+	"time"
+
+	"github.com/gruntwork-io/terratest/modules/logger"
+	"github.com/gruntwork-io/terratest/modules/terraform"
+)
+
+// Options represents the configuration options for terragrunt commands
+type Options struct {
+	// Terragrunt-specific options
+	TerragruntBinary string // The terragrunt binary to use (should be "terragrunt")
+	TerragruntDir    string // The directory containing the terragrunt configuration
+
+	// Command-specific options
+	NoColor      bool // Whether to disable colored output
+	Upgrade      bool // Whether to upgrade modules and plugins (init command)
+	Reconfigure  bool // Whether to reconfigure the backend (init command)
+	MigrateState bool // Whether to migrate state (init command)
+
+	// Configuration options
+	BackendConfig map[string]interface{} // Backend configuration (init command)
+	PluginDir     string                 // Plugin directory (init command)
+	EnvVars       map[string]string      // Environment variables for command execution
+
+	// Execution options
+	Logger                   *logger.Logger    // Logger for command output
+	MaxRetries               int               // Maximum number of retries
+	TimeBetweenRetries       time.Duration     // Time between retries
+	RetryableTerraformErrors map[string]string // Retryable error patterns
+	WarningsAsErrors         map[string]string // Warnings to treat as errors
+
+	// Terragrunt-specific extra arguments
+	ExtraArgs ExtraArgs
+}
+
+// ExtraArgs represents terragrunt-specific extra arguments for different commands
+type ExtraArgs struct {
+	Init     []string // Extra arguments for init command
+	Apply    []string // Extra arguments for apply command (used by generate)
+	Plan     []string // Extra arguments for plan command
+	Destroy  []string // Extra arguments for destroy command
+	Generate []string // Extra arguments for generate command
+}
+
+// NewOptions creates a new Options struct with default values
+func NewOptions() *Options {
```

Comment:
Function not in use, most probably we also need a function to init options for a specific directory

Reply:
Removed unused function for now. Will make them when needed. 
Hmm not too sure about the init options since I haven't seem something in the terraform options as well!